test/e2e/affinity_group.go

```diff
@@ -73,6 +73,22 @@ func AffinityGroupSpec(ctx context.Context, inputGetter func() CommonSpecInput)
 		affinityIds = executeTest(ctx, input, namespace, specName, clusterResources, "anti")
 	})
 
+	It("Should have host affinity group when affinity is soft-pro", func() {
+		cloudStackVersion := input.E2EConfig.GetVariable("CLOUDSTACK_VERSION")
```

Comment:
Can we make this dynamic and fetch the version from CloudStack itself?
The below code is not merged, but maybe something like this?

https://github.com/shapeblue/cluster-api-provider-cloudstack/blob/e96ede3956bce5df9fdb9c3306c498dd545ec07b/test/e2e/kubernetes_cks_cluster.go#L51-L68

https://github.com/shapeblue/cluster-api-provider-cloudstack/blob/e96ede3956bce5df9fdb9c3306c498dd545ec07b/test/e2e/common.go#L254-L263